python/services/dataservice/dmod/dataservice/service.py

```diff
@@ -489,6 +495,96 @@ async def _async_process_query(self, message: DatasetManagementMessage) -> Datas
         """
         return self._process_query(message)
 
+    def _build_forcing_config_for_realization(self, request: AbstractNgenRequest) -> Forcing:
+        """
+        Build a ::class:`Forcing` config object from to satisfy requirements of this request.
+
+        Function builds a ::class:`Forcing` config object as a part of the steps to create a ngen realization config
+        for the given request.  It is typically expected that the provided request will include a partial realization
+        config object that includes certain details.
+
+        Parameters
+        ----------
+        request: AbstractNgenRequest
+            An AbstractNgenRequest request that needs a realization config generated, and as part of that, a forcing
+            config.
+
+        Returns
+        -------
+        Forcing
+            Forcing config object to be used in building a ngen realization config to satisfy this request.
+        """
+        # A user may work around dataset performance issues by mounting something directly from the host into
+        # containers, using an env var for the mount source (see code in scheduler.py).  We can support this by
+        # expecting a certain prefix (from_env:::) for either the file pattern or the file basename.
+        use_from_env_workaround = False
```

Comment:
Was this mistakenly left in? I assume this was added before it was refactored into `PartialRealizationConfig`?